hooks/96-auto-manage-uefi-entries.install

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+check_if_uefi_entry_exists () {
+	for entry in $(efibootmgr); do
+
+		if [[ "$entry" == *"($efi_path"* ]] && [[ "$entry" == *"$partition_uuid"* ]]; then
+			
+			return 1
+		fi
+			
+	done
+	return 0
+}
+
+add_uefi_entry () (
+	for efi in $partition_efis; do
+		
+		# Prepare path which will be inserted into efibootmgr
+		efi_path=$(echo "${efi//$partition_mount}" | sed 's/\//\\/g')
+
+		# Add entry if it doesn't exist
+		if check_if_uefi_entry_exists; then
+
+			echo "Adding UEFI entry for $efi_path found on $partition..."
+			efibootmgr --create --disk /dev/$partition --label "Gentoo $partition" --loader "$efi_path" -u $kenrel_commands &>/dev/null
+		
+		else
+		
+			echo "UEFI Entry already exists for $efi_path on $partition !"	
+		
+		fi
+	done
+)
+
+main () {
+	efi_parttype="c12a7328-f81f-11d2-ba4b-00a0c93ec93b"
+	kernel_commands='crypt_root=UUID=dcb0cc6f-464c-4e98-b91c-e59e3655d161 root=/dev/mapper/gentoo-root rootfstype=ext4 resume=/dev/mapper/gentoo-swap dolvm quiet'
+	mounted_esp=$(lsblk -lo NAME,MOUNTPOINTS,PARTTYPE | grep  "$efi_parttype" | grep "/" | cut -d' ' -f1)
+	
+	for partition in $mounted_esp; do
+		
+		# Find partition uuid
+		partition_partuuid=$(lsblk "/dev/$partition" -lno PARTUUID)
+
+		# Find where disk is mounted
+		# Head at the end deals with cases where this partition is mounted in multiple places
+		partition_mount=$(lsblk "/dev/$partition" -lno MOUNTPOINTS | head -n 1)
+
+		# Find all .efi files on this partition
+		partition_efis="$(find $partition_mount -name "*.efi")"
```

Comment:
Does UEFI only load kernels with the `.efi` suffix? Because currently `installkernel` will only install the kernel with the `.efi` suffix if it is an Unified Kernel Image. Plain kernel images don't get any suffix, but they could in principle also be loaded directly by the UEFI (provided they are installed on the ESP).

Reply:
AFAIK yes, they need .efi extension. At least this is the case in EFI shell. Will test it.

Reply:
`ESP/EFI/Gentoo` is not a location where `installkernel` currently installs kernels though. It's `ESP/EFI/Linux/gentoo-x.y.z.efi` for UKI's (uki layout) and `ESP/gentoo/linux` for plain kernel images (bls layout). That is, if the ENTRY_TOKEN is set to `gentoo`.

Reply:
@AndrewAmmerlaan I meant it works for me without `.efi` extension.

Reply:
Yes it works even without .efi eextension. Just tested it.